### DISCUSSION TITLE
Additional Tests

### DISCUSSION THREAD

**Discussion by sharpninja** [the-benchmarker/web-frameworks#5292] [Ideas]
I propose adding FizzBuzz as an official test.  I am willing to seed several languages with implementations from https://github.com/sharpninja/FizzBuzz

**Comment by waghanza** [comment]
Not sure this is accurate to represent a real-world workload.

However, using a more accurate scenario is sommething intended before reaching maturity state. Feel free to give your feedback / opinion on what it is a real-world workload and how can it be tested.

**Reply by sharpninja** [reply]
Short of going to a database, it is actually a decent workload.  

- Recursion
- Operator completeness
- String handling
- Number Parsing

Those are all things that differentiate stacks.

**Reply by sharpninja** [reply]
More good tests

- convert json to xml (Utf16LE)
- convert json to yaml
- convert xml (Utf16LE) to yaml
- convert xml (Utf16LE) to json
- convert yaml to xml (Utf16LE)
- convert yaml to json

**Reply by sharpninja** [reply]
More good tests:

- encrypt Json Token with key pair
- decrypt Json Token with key pair
- create zip file
- hash zip file
- validate zip file
- expand zip file

Data

- convert domain object to entity and persist to json.
- load json collection, query for entity, convert result to domain model and return it.

**Reply by waghanza** [reply]
This subject is very complex. In order to be sure everyone (that will be involved) could easily understand, could you start a PR just with an `OpenAPI` documenting all of this ?

(Il test this spec to ensure all implementation respect this after we have an agreement)

**Comment by waghanza** [comment]
I've opened a new discussion in https://github.com/the-benchmarker/web-frameworks/discussions/8093 to make sure to test all features of backend tool.

Feel free to take a look, and suggest any relevant ideas @sharpninja
